azimuth_caas_operator/operator.py

```diff
@@ -115,11 +115,22 @@ async def cluster_type_updated(body, name, namespace, labels, **kwargs):
         await _update_cluster_type(K8S_CLIENT, name, namespace, cluster_type.status)
 
 
+@kopf.on.resume(registry.API_GROUP, "cluster")
+async def cluster_resume(body, name, namespace, **kwargs):
+    LOG.debug(f"Resuming cluster {name} in {namespace}")
+    # Ensure that the clusterID is set for all clusters
+    cluster = cluster_crd.Cluster(**body)
+    await cluster_utils.ensure_cluster_id(K8S_CLIENT, cluster)
+
+
 @kopf.on.create(registry.API_GROUP, "cluster")
 async def cluster_create(body, name, namespace, labels, **kwargs):
     LOG.debug(f"Attempt cluster create for {name} in {namespace}")
     cluster = cluster_crd.Cluster(**body)
 
+    # Before doing anything, ensure the clusterID is set
+    await cluster_utils.ensure_cluster_id(K8S_CLIENT, cluster)
```

Comment:
Somehow this seems clunky. Could we not do this inside the existing update_cluster logic?

Reply:
It needs to be done before because the configmap containing the extra vars is generated before `update_cluster` is ever called

Reply:
Also, I quite like having it as a distinct operation that always happens first and is guaranteed to have completed successfully before anything else runs.

Additionally, we need to be able to call just this logic in the resume handler.